docs/dev-environment-setup.md

```diff
@@ -0,0 +1,233 @@
+
+## 7.5 Bypass Authentication in Development
+
+There are some shortcuts to bypass authentication in development and run the server while logged in as a preset user:
+
+```
+cd app && yarn dev <AS_USER>
+```
+...where `<AS_USER>` is one of the following:
+
+```
+AS_REPORTER
+AS_CERTIFIER
+AS_ANALYST
+AS_ADMIN
+```
+
+[Learn more here](https://github.com/bcgov/cas-ciip-portal/blob/develop/docs/all-views-inventory.md) about the functionality available to each user type.
+
+## 8. Run Tests
+
+### Unit Tests with Jest
+
+```
+cd app && yarn test
+```
+Front-end unit tests are heavily snapshot-based. Work that changes the DOM will result in a diff from the last accepted snapshot and cause related tests to fail. You can update the snapshots and review / accept the diff with `yarn test -u`.
+
+### End-to-end Tests with Cypress
+
+```
+cd app && yarn cypress
+```
+
+### Database Unit Tests with pgTAP
+
+```
+cd schema && make unit
+```
+
+## 9. Tidy Your Work
+
+[pre-commit](https://pre-commit.com/) runs a variety of formatting and lint checks configured in [`.pre-commit-config.yaml`](https://github.com/bcgov/cas-ciip-portal/blob/develop/.pre-commit-config.yaml) which are required for a pull request to pass CircleCI.
+
+`pre-commit install` will [configure a pre-commit hook to run before every commit](https://pre-commit.com/#usage); alternatively, you can run it manually with:
+
+```
+pre-commit run --all-files
+```
+
+If you are impatient and your work is isolated to Javascript, it may be faster to run only the related formatter (`xo`), but it may not catch everything (such as the end-of-file fixer and trailing whitespace):
+
+```
+yarn format
+```
+
+<!-- ### Start & configure psql server
+- https://www.postgresql.org/download/macosx/ for an installer
+- `>$ pg_ctl start`
+- `>$ createdb <username>` where the username is your username on this mac (usually your IDIR, lowercase) (!! Important !!)
+- `>$ psql`
+    - `psql$ create role postgres with LOGIN SUPERUSER CREATEROLE CREATEDB REPLICATION BYPASSRLS`
+
+### Setup PERL env - DO NOT RELY ON SYSTEM PERL
+- make sure the latest xcode command line tools are installed (https://developer.apple.com/download/more/?=command%20line%20tools)
+- make sure homebrew is installed
+- `>$ brew install perl` ( or this https://learn.perl.org/installing/osx.html without homebrew )
+- you might have to configure perl env variables to point at local libraries (see here https://apple.stackexchange.com/questions/89271/how-to-install-modules-from-cpan-without-sudo)
+- And, make sure the PERL5LIB environment variable is properly set (wonky in my case)
+
+```
+~/.bashrc:
+    export PERL5LIB=~/lib/perl5/lib/perl5/
+
+    PERL_LOCAL_LIB_ROOT="/Users/pbastian/lib/perl5${PERL_LOCAL_LIB_ROOT:+:${PERL_LOCAL_LIB_ROOT}}"; export PERL_LOCAL_LIB_ROOT;
+    PERL_MB_OPT="--install_base \"/Users/pbastian/lib/perl5\""; export PERL_MB_OPT;
+    PERL_MM_OPT="INSTALL_BASE=/Users/pbastian/lib/perl5"; export PERL_MM_OPT;
+```
```

Comment:
I was at a bit of a loss to explain the other Perl env variables (other than `PERL5LIB`) - @pbastia can you confirm these were necessary and/or do you have a link to StackOverflow or somewhere this solution was described?

Reply:
These are some local::lib configuration variables, https://metacpan.org/pod/release/GETTY/local-lib-1.006007/lib/local/lib.pm

Reply:
OK. I recall that part of the issue you were debugging here is that your newly installed Perl was installed in a folder nested within perl itself? `~/lib/perl5/lib/perl5/`

Do we know now how it got into that state? 

Just trying to figure out if/how we should generalize this advice and whether it's likely they will need it.

Reply:
I think homebrew installed perl in `~/lib/perl5/` and then local::lib created its own `./lib/perl5/` folder in the local perl installation directory. CPAN manages this, so as long as it can find the installed modules things fit together I believe